app/config/version.py

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env python
+import subprocess
+
+# Note that this file is overwritten during the build
+# process by either the git tag of the commit
+# (see Dockerfile for details)
+
+proc = subprocess.Popen(["git", "describe", "--tags"],
+                        stdout=subprocess.PIPE,
+                        stderr=subprocess.PIPE)
+stdout, stderr = proc.communicate()
+git_version = stdout.decode('utf-8').strip()  # pylint: disable=invalid-name
+if git_version == '':
+    # If theres no git tag found in the history
+    proc = subprocess.Popen(["git", "rev-parse", "--short", "HEAD"],
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE)
+    stdout, stderr = proc.communicate()
+    APP_VERSION = f"v_{stdout.decode('utf-8').strip()}"
+else:
+    APP_VERSION = git_version
```

Comment:
I would comment both case with the output of the git command that we are expecting.
`git describe --tags` => if current commit is tagged then return tag, else return last tag found suffixed with the short hash (e.g. v0.2.0-beta.1-2-g17f5fcc)
When no tag is found uses the current git commit short hash (v_78fec8d)
This way it clearer what version we expect without knowing perfectly these git commands that most people don't know and don't use

Reply:
I added some comments.